sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -317,30 +315,18 @@ def scheduled_enqueue_time_utc(self, value):
 
     @property
     def body(self):
-        # type: () -> Any
-        """The body of the Message. The format may vary depending
-        on the body type:
-        For ~azure.servicebus.AMQPMessageBodyType.DATA, the body could be bytes or Iterable[bytes]
-        For ~azure.servicebus.AMQPMessageBodyType.SEQUENCE, the body could be List or Iterable[List]
-        For ~azure.servicebus.AMQPMessageBodyType.VALUE, the body could be any type.
-
-        :rtype: Any
+        # type: () -> Optional[Union[bytes, Iterable[bytes]]]
+        """The body of the Message.
+        :rtype: bytes or Iterable[bytes]
         """
-        return self.message.get_data()
+        # pylint: disable=protected-access
+        if not self.message._message or not self.message._body:
+            return None
 
-    @property
-    def body_type(self):
-        # type: () -> Optional[AMQPMessageBodyType]
-        """The body type of the underlying AMQP message.
+        if self.message._body.type != uamqp.MessageBodyType.Data:
+            return None  # TODO: TBD, raise TypeError vs return None
```

Comment:
```python
if message.body == None:
    message.raw_amqp_message.body

# vs

try:
    message.body
except TypeError:
    message.raw_amqp_message.body
```

although raising error on property seems not common in python, but it seems be stronger indicator to me.

there're some discussions:
https://stackoverflow.com/questions/48778945/by-design-should-a-property-getter-ever-throw-an-exception-in-python
https://softwareengineering.stackexchange.com/questions/16646/is-throwing-an-exception-from-a-property-bad-form

Reply:
I wasn't there during the original discussion, but looking through these questions, it seems that raising an error is not preferred. Is there a specific scenario you have in mind where the try/except would work and the `==` wouldn't?

Reply:
Having a case in mind which is pushing me to the `return None` approach:

 corner case for sending None:
```
sb_msg = ServiceBusMessage(body=None, application_properties={"key": "value"})
amqp_msg = AMQPAnnotatedMessage(value_body=None, application_properties={"key": "value"})
```

- if we want to adopt the raise error approach
  - our current sbmessage supports passing None (although it's not documented)
    - there is a reason, normal users might just want to send some app prop without a body
  - if users sending amqpannotatedmessage with None value body, and on the receiving side we didn't raise TypeError (because we can't differentiate)
    - then this feels like we're breaking the consistency with sequence, and non-None value body


(well, actually I have a way to convince myself that ServiceBusMessage.body is always trying to extract the data body, so in the sequence/value body case, data body is None which returns None is fair enough)

Reply:
find one raising error sample from requests content:

https://github.com/psf/requests/blob/f6d43b03fbb9a1e75ed63a9aa15738a8fce99b50/requests/models.py#L823-L841